larky/src/main/resources/vendor/Crypto/OpenPGP/__init__.star

```diff
@@ -0,0 +1,265 @@
+load("@stdlib//larky", WHILE_LOOP_EMULATION_ITERATION="WHILE_LOOP_EMULATION_ITERATION", larky="larky")
+load("@stdlib//struct", struct="struct")
+load("@vendor//Crypto/Util/py3compat", tobytes="tobytes", bord="bord", tostr="tostr")
+load("@vendor//option/result", Error="Error")
+
+StopIterating = Error("StopIteration")
+
+def _ensure_bytes(n, chunk, g):
+    for _while_ in range(WHILE_LOOP_EMULATION_ITERATION):
+        if len(chunk) >= n:
+            break
+        next_chunk = g.next()
+        if next_chunk == StopIterating:
+            return fail("OpenPGPException: Not enough bytes")
+        # chunk += next(g)
+        chunk += next_chunk
+    return chunk
+
+def _gen_one(seq):
+    self = larky.mutablestruct(__class__='generator')
+    def __init__(seq):
+        self._sequence = seq
+        self._length = len(seq)
+        self._current = -1
+        return self
+    self = __init__(seq)
+
+    def next():
+        if self._current >= self._length-1:
+            return StopIterating
+        self._current += 1
+        return self._sequence[self._current]
+    self.next = next
+    return self
+
+def _slurp(g):
+    # bs = b''
+    s = ''
+    for chunk in g:
+        s += tostr(chunk)
+    return tobytes(s)
+
+def PushbackGenerator(g):
+    self = larky.mutablestruct(__name__='PushbackGenerator', __class__=PushbackGenerator)
+    def __init__(g):
+        # self._g = g
+        self._g = g
+        self._pushback = []
+        return self
+    self = __init__(g)
+
+    def __iter__():
+        return self
+    self.__iter__ = __iter__
+
+    def next():
+        return self.__next__()
+    self.next = next
+
+    def __next__():
+        if len(self._pushback):
+            return self._pushback.pop(0)
+        # return next(self._g)
+        return self._g.next()
+    self.__next__ = __next__
+
+    def hasNext():
+        if len(self._pushback) > 0:
+            return True
+        # try:
+        #     chunk = next(self)
+        #     self.push(chunk)
+        #     return True
+        # except StopIteration:
+        #     return False
+        chunk = self.next()
+        if chunk ==  StopIterating:
+           return False
+        self.push(chunk)
+        return True
+
+    self.hasNext = hasNext
+
+    def push(i):
+        if hasattr(self._g, 'push'):
+            self._g.push(i)
+        else:
+            self._pushback.insert(0, i)
+    self.push = push
+    return self
+
+    
+def Packet():
+    """ OpenPGP packet.
+        http://tools.ietf.org/html/rfc4880#section-4.1
+        http://tools.ietf.org/html/rfc4880#section-4.3
+    """
+
+    self = larky.mutablestruct(__name__='Packet', __class__=Packet)
+
+    def __init__(data=None):
+        for tag in Packet.tags:
```

Comment:
What is Packet.tags? Did you mean self.tags?

Reply:
It's here https://github.com/singpolyma/OpenPGP-Python/blob/master/OpenPGP/__init__.py#L1637 
gonna change it to something like Packet_tags...